### PR TITLE
table to track job runs

### LINKED ISSUE #233: Create generic job to track job runs
We can leave the specific **JobRecords tables, but I'd like to create just a generic one `JobRecords`.  Every job can record runs to that table, and we'll filter via the UI.  For starters, the new AEJournalJob will use it, but more jobs are coming soon.

Fields:
Id
JobName
StartedAt
EndedAt (nullable)
Status
Details (max length string filled with custom JSON if needed.  Used for recording processedDate for Cybersource, and maybe stuff like # records processed, generic useful metadata)

Additionally, I'd like to know which job modified a given transaction.  Currently this is done with individual properties on the Transaction object, i.e. `KfsScrubberUploadJobRecordId`.  I think it'd be clearer to make a new glue table to many-to-many Transactions to the new JobRecords table.

Then, for example, a run of AEJournalJob will write a new JobRecord with an Running status, and then if during that run it uploads a new GL or moves the transaction to completed status, etc, it should add a `jobRecord.Transactions.Add()` reference so we can look that up later.

### PR DESCRIPTION
**Pull request by srkirkland**
Basic table which contains info about job runs, both in progress and once completed.  Stores a json object with run details for later inspection.

closes #233 but we should eventually make some UX for viewing recent job runs

### CODE REVIEW

File: Sloth.Core/Domain/JobRecord.cs [hunk 1]
```diff
@@ -1,20 +1,51 @@
 using System;
-using System.Collections.Generic;
 using System.ComponentModel.DataAnnotations;
+using Newtonsoft.Json;
```
Comment (jSylvestre) [inline Sloth.Core/Domain/JobRecord.cs:3]: We are still using Newronsoft?

File: Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs [hunk 2]
```diff
@@ -23,8 +24,35 @@ public AggieEnterpriseJournalJob(SlothDbContext context, IAggieEnterpriseService
...
+            public AggieEnterpriseJournalJobDetails()
+            {
+                TransactionRunStatuses = new List<TransactionRunStatus>();
+            }
+
+            public List<TransactionRunStatus> TransactionRunStatuses { get; set; }
+
+            public int TransactionsProcessedCount => TransactionRunStatuses.Count;
+
+            public class TransactionRunStatus
```
Comment (jSylvestre) [inline Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs:38]: Did you want this outside the class?

File: Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs [hunk 3]
```diff
@@ -85,6 +119,9 @@ public async Task UploadTransactions(ILogger log)
 
                                 journalRequest.RequestId = requestStatus.RequestId.Value;
                                 journalRequest.Status = requestStatus.RequestStatus.ToString();
+
+                                transactionRunStatus.Action = requestStatus.RequestStatus.ToString();
+                                ;
```
Comment (jSylvestre) [inline Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs:124]: trailing ;

**Review by jSylvestre — APPROVED** [review]
Didn't run anything, but other than a few comments I don't see any problems.

**Review by sprucely — APPROVED** [review]

File: Sloth.Core/Domain/JobRecord.cs (hunk 1 above)
Comment (srkirkland) [reply Sloth.Core/Domain/JobRecord.cs:3]: I think so in sloth but i can double check

File: Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs (hunk 2 above)
Comment (srkirkland) [reply Sloth.Core/Jobs/AggieEnterpriseJournalJob.cs:38]: I was thinking no, but if you can think of a use case we can extract it.

File: Sloth.Core/Domain/JobRecord.cs (hunk 1 above)
Comment (srkirkland) [reply Sloth.Core/Domain/JobRecord.cs]: I changed it to use system.text.

**Pull request merged**